src/send_packets.c

```diff
@@ -1211,3 +1279,48 @@ get_user_count(tcpreplay_t *ctx, sendpacket_t *sp, COUNTER counter)
 
     return(uint32_t)send;
 }
+#ifdef HAVE_LIBXDP
+static inline void fill_umem_with_data_and_set_xdp_desc(sendpacket_t* sp, int tx_idx, COUNTER umem_index, u_char* pktdata, int len){
+    COUNTER umem_index_mod = (umem_index % sp->batch_size) * sp->frame_size; // packets are sent in batch, after each batch umem memory is reusable
```

Comment:
this line is responsible for reusable UMEM memory. Due to the fact that we are sending packets in batches and after each of the batches we are waiting until all the packets have been transmitted in a given batch, we only need batch_size * frame_size memory

Reply:
Yes, thats fine. Just make sure we have enough umem allocated for a batch!